android/src/main/java/io/invertase/firebase/database/RNFirebaseDatabaseReference.java

```diff
@@ -346,8 +348,13 @@ private void handleDatabaseError(ReadableMap registration, DatabaseError error)
    * @return
    */
   private void buildDatabaseQueryAtPathAndModifiers(String path, ReadableArray modifiers) {
-    FirebaseApp firebaseApp = FirebaseApp.getInstance(appName);
-    FirebaseDatabase firebaseDatabase = FirebaseDatabase.getInstance(firebaseApp);
+    FirebaseDatabase firebaseDatabase;
```

Comment:
It would make sense to increase the visibility of and make use of the `getDatabaseForApp` method on RNFirebaseDatabase.

Reply:
Since RNFirebaseDatabaseReference doesn't have an instance of RNFirebaseDatabase and getDatabaseForApp modifies and gets values from its instance variables, I can't make that method static or call it from RNFirebaseDatabaseReference. If its ok I can extract the FirebaseDatabase initialization logic from that method and create a new method that also handles logging and call that from inside RNFirebaseDatabase.